client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx

```diff
@@ -199,8 +202,16 @@ const LicensingActivationThankYou: FC< Props > = ( {
 	] );
 
 	const siteSelectOptions = useMemo( () => {
+		const isProductActivatedOnSite = ( product: Product ) =>
+			product && product.product_slug === productSlug;
+
 		return jetpackSites
 			.filter( ( site: JetpackSite ) => ! site.is_wpcom_atomic )
+			.filter(
+				( site: JetpackSite ) =>
+					! site.products.some( isProductActivatedOnSite ) &&
+					! isProductActivatedOnSite( site.plan )
+			)
```

Comment:
You don't need to set the type for `site` inside filter callback. TS is intelligent enough to do it for you. When you use `.map()` or `.filter()` on an array of certain type, the callback types are automagically set. 👇 

<img width="547" alt="image" src="https://user-images.githubusercontent.com/18226415/148538084-068f8346-7055-4e46-9410-f3cd64612423.png">

Reply:
Thanks for making note of this! Do you think there's any benefit of explicitly specifying the type? Probably most of the modern IDEs will display the type for (as you shown), I'm just curious why we have listed them explicitly before.

Reply:
I don't think there is any benefit. My understanding is that the recommendation is to try to rely as much as possible on TS' capabilities to implicitly detect a type.

> I'm just curious why we have listed them explicitly before.

I don't know the real reason, but it could be that since TS is constantly evolving and getting more intelligent, maybe when this was implemented, TS wasn't capable of detecting the type of `site` without help.

Reply:
Thanks! Fixed in 854e1fe4b11f0c855be89982b7c05bd018af01f2